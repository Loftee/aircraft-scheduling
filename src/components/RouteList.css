.route-list-wrapper {
    width: 37.5%;
    padding: 15px;
    float: left;
    list-style: none;
    margin: 0;
}

.route-list {
    width: 100%;
    float: left;
    list-style: none;
    padding: 0;
}

.route-list-item {
    width: 100%;
    margin-bottom: 20px;
    padding: 10px;
    border: 1px solid #ccc;
    cursor: pointer;
}

.route-list-item:hover {
    background: #eee;
    border-color: #999;
}

.route-list-item__data {
    font-size: 0;
}

.route-list-item__data span {
    font-size: initial;
    display: inline-block;
    width: 40%;
    vertical-align: middle;
    min-height: 20px;
}

.route-list-item__data span:last-child {
    text-align: right;
}

span.route-list-item__data-arrow {
    width: 20%;
    text-align: center;
    background: url('../assets/arrow.png') center center no-repeat;
    background-size: contain;

}

.route-list-reset {
    float: left;
    width: 100%;
    text-align: right;
}