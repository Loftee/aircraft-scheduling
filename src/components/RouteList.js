import React, {Component} from 'react';

import './RouteList.css';

class RouteList extends Component {
    render() {
        const {routeList} = this.props;
        // console.log(this.props);
        return (
            <div className={'route-list-wrapper'}>
                <h2>Routes</h2>
                <ul className={'route-list'}>
                    {
                        routeList.map((route, index) => {
                            return (
                                <li
                                    key={index}
                                    className={'route-list-item'}
                                >
                                    <h3 className={'route-list-item__header'}>Flight: {route.id}</h3>
                                    <div className={'route-list-item__data'}>
                                        <span>Departure: {route.readableDeparture} - {route.origin}</span>
                                        <span className={'route-list-item__data-arrow'}/>
                                        <span>Arrival: {route.readableArrival} - {route.destination}</span>
                                    </div>
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        )
    }
}

export default RouteList;