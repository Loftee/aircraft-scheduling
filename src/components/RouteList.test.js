import React from 'react';
import {shallow} from 'enzyme';
import RouteList from './RouteList';

describe('Route List', () => {
    const props = {
        routeList:
            [{
                id: "AS1001",
                departureTime: 21600,
                arrivalTime: 26100,
                readableDeparture: "06:00",
                readableArrival: "07:15",
                origin: "LFSB",
                destination: "LFMN"
            }]
    };
    const component = shallow(<RouteList {...props} />);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('displays selected routes', () => {
       expect(component.find('.route-list-item').length).toBe(1);
    });
});