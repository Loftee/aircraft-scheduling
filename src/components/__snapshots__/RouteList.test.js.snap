// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route List Render correctly 1`] = `
<ul
  className="route-list"
/>
`;

exports[`Route List renders correctly 1`] = `
<div>
  <h2>
    Routes
  </h2>
  <ul
    className="route-list"
  >
    <li
      className="route-list-item"
      key="0"
    >
      <h3
        className="route-list-item__header"
      >
        Flight: 
        AS1001
      </h3>
      <div
        className="route-list-item__data"
      >
        <span>
          Departure: 
          06:00
           - 
          LFSB
        </span>
        <span
          className="route-list-item__data-arrow"
        />
        <span>
          Arrival: 
          07:15
           - 
          LFMN
        </span>
      </div>
    </li>
  </ul>
</div>
`;
